### PR TITLE
fix: Added deprecated functions notice & handled backward compatibility.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
fix: Added deprecated functions notice & handled backward compatibility.

### Types of changes
Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
 - All the deprecated functions used in 3rd party themes/plugins should work as it is.

### Checklist:
- [x] My code is tested
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-helper.php [hunk 1]
```diff
@@ -975,6 +975,50 @@ public static function generate_all_css( $combined_selectors, $id ) {
...
+		/**
+		 * Parse Guten Block.
+		 *
+		 * @since 1.1.0
+		 * @param string $content the content string.
+		 * @deprecated 1.23.0 Use `UAGB_Frontend::parse_blocks()` instead
+		 * @access public
+		 */
+		public function parse( $content ) {
+
```
Comment (Nikschavan) [inline classes/class-uagb-helper.php:1017]:
```suggestion
```

File: classes/class-uagb-helper.php [hunk 2]
```diff
@@ -975,6 +975,50 @@ public static function generate_all_css( $combined_selectors, $id ) {
...
+		/**
+		 * Generates stylesheet for reusable blocks.
+		 *
+		 * @since 1.1.0
+		 * @param array $blocks Blocks.
+		 * @deprecated 1.23.0 Use `UAGB_Frontend::get_blocks_assets()` instead
+		 * @access public
+		 */
+		public function get_assets( $blocks ) {
+
```
Comment (Nikschavan) [inline classes/class-uagb-helper.php:1002]:
```suggestion
```

File: classes/class-uagb-helper.php [hunk 3]
```diff
@@ -975,6 +975,50 @@ public static function generate_all_css( $combined_selectors, $id ) {
...
+		/** Generates stylesheet in loop.
+		 *
+		 * @since 1.7.0
+		 * @param object $this_post Post Object.
+		 * @deprecated 1.23.0 Use `UAGB_Frontend::prepare_assets()` instead
+		 * @access public
+		 */
+		public function get_generated_stylesheet( $this_post ) {
+
```
Comment (Nikschavan) [inline classes/class-uagb-helper.php:987]:
```suggestion
```

**Review by Nikschavan — APPROVED** [review]
👍

Although I think we should have a deprecation policy. Any function/method/class should be deprecated first and then after a fixed number of stable releases we remove/rename it. This will avoid any accidental breakages because any third party plugin has not been updated on the site which uses older API from the plugin

**Review by vrundakansara — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**